Apps/BMSQueryApp/app/src/main/java/uk/ac/cam/cares/jps/bmsqueryapp/view/tab/EditFragment.java

```diff
@@ -74,22 +76,62 @@ public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceStat
             try {
 //                Double temperatureDouble = Double.parseDouble(temperature);
                 JSONObject params = new JSONObject();
-                params.put("dataIRI", editableAttributes.get(0).getIri());
-                params.put("temperature", Double.parseDouble(editableAttributes.get(0).getValue()));
-                params.put("clientProperties", "CLIENT_PROPERTIES");
-
-                JsonObjectRequest jsonObjectRequest = new JsonObjectRequest(Request.Method.POST, ESPHOME_CONTROL_URL.build().toString(), params, response -> {
-                    try {
-                        String fanStatus = response.getString("fanStatus");
-                        Toast.makeText(this.getContext(), fanStatus.isEmpty() ? fanStatus : "Successfully updated.", Toast.LENGTH_LONG).show();
```

Comment:
This toast may give wrong info to the users. To change the setpoint value, both control mode and setpoint update request should be successful. The current way only ensures the setpoint update request succeeded. Since the two requests are handle by the same agent, it is ok to call the two request nested to ensure that both succeeded.
``` java
request1 (..., response -> {
    request2 = (..., response -> {
        Toast
    )
    SingletonConnection.getInstance(this.getContext()).addToRequestQueue(request2);
)

SingletonConnection.getInstance(this.getContext()).addToRequestQueue(request1);
```

Reply:
Nested both requests. The code will write values to the editable variable first before attempting to change the control mode.